doc/src/SUMMARY.md

```diff
@@ -0,0 +1,20 @@
+# Summary
+
+[Where to start](ch00-00-where-to-start.md)
+
+- [How an event loop works](ch01-00-how-an-event-loop-works.md)
+- [A simple example](ch02-00-a-simple-example.md)
+- [A full example: ZeroMQ](ch03-00-a-full-example-zeromq.md)
+  - [Composition](ch03-01-composition.md)
+  - [Creating our source, part I: our types](ch03-02-creating-our-source-part-1-our-types.md)
+  - [Creating our source, part II: setup methods](ch03-03-creating-our-source-part-2-setup-methods.md)
+  - [Creating our source, part III: processing events (almost)](ch03-04-creating-our-source-part-3-processing-events-almost.md)
+  - [Creating our source, part IV: processing events (really)](ch03-05-creating-our-source-part-4-processing-events-really.md)
+  - [The full ZeroMQ event source code](ch03-06-the-full-zeromq-event-source-code.md)
+- [How to do X in Calloop](ch04-00-how-to-do-x-in-calloop.md)
```

Comment:
As a general organization, it feels to me that "How to do X" should come before "How to write my custom event source" in the chapter list.

Reply:
I'd be happy with that, but I'll do at it the end of the edits to keep the diff relatively clean.